cmd/influx/main.go

```diff
@@ -1,26 +1,146 @@
 package main
 
 import (
+	"context"
+	"crypto/tls"
 	"fmt"
+	"net/http"
 	"os"
+	"runtime"
 	"time"
 
+	"github.com/influxdata/influx-cli/v2/api"
+	"github.com/influxdata/influx-cli/v2/internal/config"
+	"github.com/influxdata/influx-cli/v2/internal/ping"
+	"github.com/influxdata/influx-cli/v2/pkg/tracing"
 	"github.com/urfave/cli/v2"
 )
 
+// Fields set via ldflags at build time.
 var (
 	version = "dev"
 	commit  = "none"
 	date    = ""
 )
 
+var (
+	tokenFlag      = "token"
+	hostFlag       = "host"
+	skipVerifyFlag = "skip-verify"
+	traceIdFlag    = "trace-debug-id"
+	configPathFlag = "config-path"
+	configNameFlag = "active-config"
+)
+
+// loadConfig reads CLI configs from disk, returning the config with the
+// name specified over the CLI (or default if no name was given).
+func loadConfig(ctx *cli.Context) (config.Config, error) {
+	configs := config.GetConfigsOrDefault(ctx.String(configPathFlag))
+	configName := ctx.String(configNameFlag)
+	if configName != "" {
+		if err := configs.Switch(configName); err != nil {
+			return config.Config{}, err
+		}
+	}
+	return configs.Active(), nil
+}
+
+// newApiClient returns an API client configured to communicate with a remote InfluxDB instance over HTTP.
+// Client parameters are pulled from the CLI context.
+func newApiClient(ctx *cli.Context, injectToken bool) (api.ClientInterface, error) {
```

Comment:
I've manually tested that the `host`-related pieces of this work. It's hard to test that the auth-token things do what we want when testing against `/ping`; there will probably need to be follow-up fixes.